tsdb/head.go

```diff
@@ -472,7 +516,11 @@ func (h *Head) loadWAL(r *wal.Reader, multiRef map[uint64]uint64) (err error) {
 			for _, s := range v {
 				series, created := h.getOrCreateWithID(s.Ref, s.Labels.Hash(), s.Labels)
 
-				if !created {
+				if created {
+					if mc, ok := mmappedChunks[series.ref]; ok {
+						series.mmappedChunks = mc
+					}
```

Comment:
I have a question here: Consider the series was not created and it was an old one, should the mmapped chunks be taken for `series.ref` (already present series) or from `s.Ref` (the reference from new series record), as the reference of the series active before shutting down would be from the last series record.

Reply:
Or more like it is possible we need to club the mmapped chunks for both `series.ref` and `s.Ref`

Reply:
So I have kept it as a TODO for the future as there are some races to take care when we are removing the old data when we come across a duplicate record. As for now, the WAL replay will take care of adding back all the data and Head gc will take care of removing old data at the end.